shared/src/commonMain/kotlin/org/hyperskill/app/analytic/data/repository/AnalyticHyperskillRepositoryImpl.kt

```diff
@@ -24,6 +26,12 @@ internal class AnalyticHyperskillRepositoryImpl(
             eventsToFlush = hyperskillCacheDataSource.getEvents()
             hyperskillCacheDataSource.clearEvents()
         }
-        return hyperskillRemoteDataSource.flushEvents(eventsToFlush, isAuthorized)
+        return hyperskillRemoteDataSource
+            .flushEvents(eventsToFlush, isAuthorized)
+            .onFailure {
+                mutex.withLock {
+                    hyperskillCacheDataSource.logEvents(eventsToFlush)
```

Comment:
I suppose that there is a case where the current user profile can be changed between `logEvent`, and old analytic events can revert to being associated with the previous user profile.
However, it is not so critical right now.

Reply:
@ivan-magda 
Yes, you are right. Good point to think about.